Update Network CIM Analytics With FTD Data

nasbench opened the pull request:
The following PR maps some of Network CIM analytics we have to the Cisco datasources and providing test data. Below is the full list of updated analytics:

- Detect Outbound LDAP Traffic
- Detect Outbound SMB Traffic
- Internal Horizontal Port Scan
- Internal Horizontal Port Scan NMAP Top 20
- Internal Vertical Port Scan
- Prohibited Network Traffic Allowed
- Protocol or Port Mismatch
- Protocols passing authentication in cleartext
- TOR Traffic

Due to the nature of the mapping with the CSC TA. The Analytic `Detect Remote Access Software Usage Traffic` cannot be used as is. The current config is mapping the `Application` field to the CIM `app` field. Which is technically accurate. But in order to use this detection as is, we need to map the `ClientApplication` field which does not have an equivalent in CIM.

For this a new equivalent analytic has been created `Cisco Secure Firewall - Remote Access Software Usage Traffic` along with a dedicated lookup that is directly taken from Cisco Secure Firewall App Detectors list.

nasbench committed:
update network cim analytics with ftd data

nasbench committed:
Merge branch 'develop' into ftd-cim-update

nasbench committed:
update output fields

nasbench committed:
field order update

nasbench committed:
test more network analytics with ftd

nasbench committed:
add drilldowns

nasbench committed:
add missing output fields

nasbench committed:
ci fixes

nasbench committed:
Update protocols_passing_authentication_in_cleartext.yml

nasbench committed:
Update tor_traffic.yml

nasbench committed:
Update tor_traffic.yml

nasbench committed:
update how to implement section

nasbench committed:
add `Cisco Secure Firewall - Remote Access Software Usage Traffic`

nasbench committed:
Update cisco_secure_firewall___remote_access_software_usage_traffic.yml

nasbench committed:
Merge branch 'develop' into ftd-cim-update

pyth0n1c commented on detections/network/protocols_passing_authentication_in_cleartext.yml:
```diff
@@ -13,22 +13,37 @@ description: The following analytic identifies the use of cleartext protocols th
+  earliest_offset: $info_min_time$
+  latest_offset: $info_max_time$
 rba:
   message: Potential Authentication in cleartext
```
I think we should probably include $dest_port$ here as it gives more information about the specific protocol (assuming default port) which is triggering the alert. I realize this change was not made as part of this PR.

nasbench replied:
I changed the RBA message to `Allowed Traffic from $src_ip$ to $dest_ip$ over port $dest_port$. Which might indicate a potential authentication attempts over a  cleartext protocol.`. Which is better suited for the type of alert.

pyth0n1c commented on detections/network/protocols_passing_authentication_in_cleartext.yml:
```diff
@@ -13,22 +13,37 @@ description: The following analytic identifies the use of cleartext protocols th
-data_source: []
+data_source:
+- Cisco Secure Firewall Threat Defense Connection Event
 search: '| tstats `security_content_summariesonly` count min(_time) as firstTime max(_time)
```
Should we add an eval / IF or something similar (a lookup is probably too heavy) to map the specific ports in question to a service?
Or do we just assume users know what all of these services are?

nasbench replied:
As an improvement to the rule yes. This could be extended with a lookup containing a list of clear text protocols, default port, and some additional descriptions for better analysis.

I think this is better served as an update in the next release instead of this one. As this is release week.

pyth0n1c commented on detections/network/internal_vertical_port_scan.yml:
I think src_ip should probably be a threat_object and dest_ip should probably be a risk_object.

pyth0n1c commented on detections/network/internal_horizontal_port_scan_nmap_top_20.yml:
Should also likely have both threat (src) and risk (dest_zone, maybe, since we do not have explicit dest_ip?) objects

pyth0n1c commented on detections/network/internal_horizontal_port_scan.yml:
Risk/threat objects again? Maybe since there is no dest_ip (presumably since there would be a huge number) it's actually the port that would be the risk_object?

pyth0n1c reviewed (changes requested):
No actual issues and all the tests + results were inspected and look good.
I do want to get some feedback, however, before approving and merging
with respect to some of the RBA fields, mostly around threat and risk objects.
Those fields were not actually updated as part of this PR, but since I took a 
look at them I want to make sure they are in line with our expectations.

nasbench committed:
update analytics after review suggestions

nasbench committed:
Merge branch 'develop' into ftd-cim-update

nasbench committed:
Update protocols_passing_authentication_in_cleartext.yml

nasbench committed:
update risk object

pyth0n1c committed:
Update internal_horizontal_port_scan_nmap_top_20.yml

Chatted with detection author about this update.
Changed risk_object from dest_zone to dest_ports to be more in line with other horizontal scanning detection AND give more meaningful risk object (dest_zone is less meaningful, even if populated)

pyth0n1c reviewed (approved):
After discussions with author and one last updated, approving this PR.
We will make further changes in the future, but these changes are a significant improvement for now.
